### PR TITLE
Change language

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -33,17 +33,19 @@ const messages = {
...
-        <Provider store={store}>
-            <BrowserRouter basename={basename}>
-                <App />
-            </BrowserRouter>
-        </Provider>
-    </IntlProvider>,
-    document.getElementById('root')
-);
+const ApplicationWrapper = () => {
+    const [language, setLanguage] = useState();
```
Comment (ChamseddineBhd) [inline src/index.js:39]: what is the intial value of this state !  you should initialize it with SYSTEM ?

File: src/components/app.js [hunk 2]
```diff
@@ -395,6 +416,11 @@ const App = () => {
         updateConfigParameter(PARAMS_USE_NAME_KEY, useName);
     };
 
+    const handleLanguageClick = (language) => {
```
Comment (ChamseddineBhd) [inline src/components/app.js:419]: you should also dispatch the new value to the store

**Review by ChamseddineBhd — CHANGES_REQUESTED** [review]

File: src/components/app.js [hunk 3]
```diff
@@ -157,13 +163,24 @@ const noUserManager = { instance: null, error: null };
 
 const STUDY_VIEWS = [StudyView.MAP, StudyView.SPREADSHEET, StudyView.RESULTS];
 
-const App = () => {
+const supportedLanguages = [FRENCH, ENGLISH];
+
+const getSystemLanguage = () => {
```
Comment (jonenst) [inline src/components/app.js:168]: rename to getComputedLanguage or something

File: src/index.js (hunk 1 above)
Comment (jonenst) [reply src/index.js:39]: there should not be a state here since the language is stored in redux

File: src/components/app.js (hunk 3 above)
Comment (etiennehomer) [reply src/components/app.js:168]: This method is really to get the language of the system. There's already a method named getComputedLanguage wheter it is a 'sys' code or a language code ('en', 'fr', etc)

File: src/components/app.js (hunk 2 above)
Comment (jonenst) [reply src/components/app.js:407]: The value is dispatched to the store when the websocket notifies that the server has stored the new value. All configs behave this way. This PR should not change that so the code is correct like this I think

File: src/redux/reducer.js [hunk 4]
```diff
@@ -55,6 +57,8 @@ const initialState = {
     selectedCase: null,
     selectedFile: null,
     useName: true,
+    language: 'sys',
+    computedLanguage: 'en',
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/redux/reducer.js]: Done

**Review by jonenst — APPROVED** [review]

**Pull request merged**